From c4ba7a8f6ff7f8ee1ad2d5fa3fc5f6bc5f495300 Mon Sep 17 00:00:00 2001
From: Yann Dirson <yann.dirson@vates.fr>
Date: Wed, 7 Feb 2024 10:23:57 +0100
Subject: [PATCH] Prevent upgrading from platform 3.4.0

(or even 3.3.0, which was the CH preview and XCP-ng 8.3 alpha)

This is a hack/workaround to be removed in the next 8.3 beta, once
we've aligned user handling with XS8.

Signed-off-by: Yann Dirson <yann.dirson@vates.fr>
---
 upgrade.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/upgrade.py b/upgrade.py
index 4c3b887..d2ee14c 100644
--- a/upgrade.py
+++ b/upgrade.py
@@ -187,7 +187,10 @@ class Upgrader(object):
 class ThirdGenUpgrader(Upgrader):
     """ Upgrader class for series 7+ Retail products. """
     upgrades_product = version.PRODUCT_NAME
-    upgrades_versions = [ (product.XENSERVER_MIN_VERSION, product.THIS_PLATFORM_VERSION) ]
+    # XCP-ng 8.3beta2 only: prevent upgrade from XS8 and 8.3 iself to
+    # workaround the "missing certusers" issue, to be fixed in beta3
+    max_platform = Version([3, 2, 1]) # we and XS8 are 3.4.0
+    upgrades_versions = [ (product.XENSERVER_MIN_VERSION, max_platform) ]
     upgrades_variants = [ 'Retail' ]
     requires_backup = True
     optional_backup = False
-- 
2.39.2

